qiskit_experiments/experiment_data.py

```diff
@@ -364,5 +364,7 @@ def __str__(self):
         if n_res:
             ret += "\nLast Analysis Result"
             for key, value in self._analysis_results[-1].items():
+                if key in ("raw_data", "pcov"):
```

Comment:
This can be removed if you make the change that `raw_data` is optional with default of False. These sort of displays should ideally be handled by writing a repr method for `AnalysisData` anyway.

Reply:
Added `CurveAnalysisResult` subclass to manage string representation of curve fit analysis. See 00519d0